### PR TITLE
shared-mime-info: 2.2 -> 2.3

### PR DESCRIPTION
**Pull request by tobim**
## Description of changes

https://gitlab.freedesktop.org/xdg/shared-mime-info/-/releases/2.3
Also adds a patch for clang_16 compat.

Supersedes #252576.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by tobim** [comment]
Related to #234710.

**Review by mimame — APPROVED** [review]

File: pkgs/data/misc/shared-mime-info/default.nix [hunk 1]
```diff
@@ -23,9 +23,15 @@ stdenv.mkDerivation rec {
     owner = "xdg";
     repo = pname;
     rev = version;
-    sha256 = "sha256-QrRe/DcjpTMejHXDSOLbjpJywod8qIjP6/leTZ21rhE=";
+    sha256 = "sha256-cEfknRVtOJykEO9Iqlb0UoiayYtu+ugvmmZqAD5cGnE=";
   };
 
+  patches = [
+    # Submitted upstream at
+    # https://gitlab.freedesktop.org/xdg/shared-mime-info/-/issues/211.
```
Comment (jtojnar) [inline pkgs/data/misc/shared-mime-info/default.nix:31]: Maybe remove the period here, GitHub considers it a part of the link :crying_cat_face:

**Review by jtojnar — APPROVED** [review]
[Upstream diff](https://gitlab.freedesktop.org/xdg/shared-mime-info/-/compare/2.2...2.3) looks okay (did not check the XML file).

**Comment by tobim** [comment]
The clang16 patches have been merged upstream and there is still an open MR to update the changelog and the version number: https://gitlab.freedesktop.org/xdg/shared-mime-info/-/merge_requests/250/diffs?commit_id=e24fe6127370880596b9f06a4dd4a671ad530f85

I'll set this to draft for now in case the tag will be rewritten.

**Pull request merged**